### ISSUE TITLE
Execute mutations in order

### ISSUE THREAD

**Issue by ccakes** [graphql-perl/graphql-perl#27]
https://github.com/graphql-perl/graphql-perl/blob/f1e7ad1e5a6fe071cc6e24364862706689a20b6d/lib/GraphQL/Type/Object.pm#L105-L146

In this method, because `$fields_got` is a hashref, if a user specifies multiple mutations within a query the order of execution isn't necessarily the same as the user specified.

An example I'm considering here is this:
```graphql
    mutation updateThings($foo: DeviceInput, $bars: [BarInput!]) {
      updateFoo(input: $foo) {
        id
      }

      updateBars(input: $bars) {
        id
      }
    }
```

`updateFoo` and `updateBars` end up as hashref keys and so are executed in random order.

It'd be great to provide some guarantee that the mutations are run per the input order, `$selections` maintains the correct order so it should be possible to keep this. Any thoughts on how to proceed? I'd be open to putting together a PR but wanted to make sure the current behaviour isn't intended

**Comment by mohawk2** [comment]
The current behaviour is not intended. I skated over this during the original port because I didn't realise JavaScript objects return keys in the order they're inserted. Fixed and released as 0.43. Thanks for the prompt!

**Comment by mohawk2** [comment]
By the way, on reflection - mutations (and any operation) are guaranteed to now be *initiated* in the given order. `_execute_fields_serially` is still a stub, and async operations will still all happen concurrently-ish. That function still needs filling out to do the same as the JS implementation: to do as it does now when no `PromiseCode`, but if some is provided, to execute each operation and `then` the next after it, etc. If you'd like to PR that, go ahead!
